leetcode/src/45.c

```diff
@@ -0,0 +1,50 @@
+// Breadth-first search, imitation.
+// Runtime: O(n)
+// Space: O(n)
+int jump(int* nums, int numsSize) {
+    if (numsSize == 1) {
+        return 0;
+    }
+
+    int step = 1;
+    int* visitedCells = calloc(numsSize, sizeof(int));
+
+    int* queue = malloc(numsSize * sizeof(int));
+    queue[0] = 0;
+    int queueLength = 1;
+
+    while (queueLength > 0){
+        int* nextQueue = malloc(numsSize * sizeof(int));
+        int nextQueueLength = 0;
+
+        for (int i = 0; i < queueLength; i++) {
+            int cell = queue[i];
+            int jump = nums[cell];
+
+            if (cell + jump >= numsSize - 1) {
+                free(visitedCells);
+                free(queue);
+                free(nextQueue);

```

Comment:
Non-blocking suggestion.

Whenever you have more than one resource that needs to be freed at the same time, usually that means that the implementation should guard against accidentally missing a `free` statement of one of those resources. So maybe it might benefit you if you move this `free`ing of resources into its separate function 😄

Reply:
@tjgurwara99 I thought about make a function with calculating result, but the `main` function handled free resources at the end, before return result.